language-design.md

```diff
@@ -27,7 +56,19 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
+In the sound lab there was the issue that each sound could only have a single transformation applied to it (and an unavoidable side effect of playing the sound and writing to a file).  This made it very hard to extend the library - multiple transformations couldn't be done by cleanly calling multiple methods sequentially, but instead by writing a new function that explicitly performed all of those transformations (or even worse, daisy-chaining using the fact that the output file name was "out.wav").  This violated [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) as well as ease of expansion and use.  
```

Comment:
Like all principles, dogmatic adherence to them is usually bad.  Keeping in mind that principles exist for a reason, and following them judiciously, is usually good.

That being said, I'd say that your example of semantical repetition is inaccurate - `list.sort()` sorts in-place, while `sorted(iterable)` returns a new, sorted iterable.